client/src/App.js

```diff
@@ -0,0 +1,125 @@
+import './App.css';
+import { useState } from 'react';
+import Score from './components/Score';
+import Question from './components/Question';
+import Flag from './components/Flag';
+
+function App() {
+
+  const [submitted, setSubmitted] = useState(false);
+  const [questionChosen, setQuestionChosen] = useState(null);
+  const [questionIndex, setQuestionIndex] = useState(0);
+  const [displayQuestion, setDisplayQuestion] = useState(false);
+  const [resultMessage, setResultMessage] = useState(null);
+  const [score, setScore] = useState(0);
+  const [disable, setDisable] = useState(false);
+  const [flagIndex, setFlagIndex] = useState(1);
+
+  const message = [
+    `Sorry, your answer is incorrect.`,
+    `Bravo! Your answer is correct. You are one step closer to completing the puzzle!`, 
+    `Bravo! Your answer is correct. Great job completing parts of the puzzle!`,
+    `Bravo! Your answer is correct. Congratulations! You have successfully completed the puzzle!`
+  ];
+
+  const flagPieces = document.getElementsByClassName('flag');
```

Comment:
This feels a bit funky/confusing for two reasons:

1. It's not very reacty to query elements by class name. This side steps the built in react functionality for computing diffs between prop changes to components.
2. It is actually harder to read. The HTML elements of the "flag" class are defined in another component which is kind of a nono. You shouldn't reference the raw HTML of another component. Basically a component should only change the props of HTML elements defined inside of it and if you need to trigger a change in an HTML element defined in another component, the best thing to do is to define a function in the component that you want to change, and pass that function to the component that needs to do the change. I'll try to come back and give an example.

Reply:
It's gotten a bit late, but maybe we can dig into this specifically tomorrow. Reading through your code a bit more closely, I think the "App" component should have a "FlagPieces" state object which it passes to the Flag component.